cmd/repo-updater/repoupdater/server.go

```diff
@@ -614,23 +614,12 @@ func (s *Server) computeNotClonedCount(ctx context.Context) (uint64, error) {
 		return 0, err
 	}
 
-	notCloned := make(map[string]struct{}, len(names))
-	for _, n := range names {
-		lower := strings.ToLower(string(n))
-		notCloned[lower] = struct{}{}
-	}
-
-	cloned, err := s.GitserverClient.ListCloned(ctx)
+	cloned, err := s.Store.CountClonedRepos(ctx)
```

Comment:
Since you need to `ListAllRepoNames` here and thus need to make two queries, you could maybe change your `CountClonedRepos` to return two values, so it could be used like this:

```suggestion
	cloned, uncloned, err := s.Store.CountClonedRepos(ctx)
```

Then you only have a single `SELECT`

Reply:
Actually, if this is the only place we're using this new method and what we actually need is the count of *not cloned* repos, then we should instead implement `Store.CountNotClonedRepos` and delete `Store.ListAllRepoNames`.

Reply:
Agreed, it makes sense